redfish_client/resource.py

```diff
@@ -112,3 +119,28 @@ def execute_action(self, action_name, payload):
     @property
     def raw(self):
         return self._content
+
+    def post(self, path, payload):
+        """
+        Perform a POST at the resource with the given payload.
+
+        Args:
+          path: Path to POST to.
+          payload: The contents of the POST payload.
+        """
+        return self._connector.post(path, payload=payload)
+
+    def delete(self, path):
+        """
+        Perform a DELETE at the resource.
+
+        Args:
+          path: The path to the resource to DELETE.
+        """
+        return self._connector.delete(path)
+
+    def refresh(self):
+        oid = self._content.get("@odata.id")
```

Comment:
Silently ignoring the missing `@odata.id` might bite us in the future. Maybe raising some kind of exception if the refresh cannot be done would be a bit better since it would offer our end-users to detect the failure? Adding an *I-know-what-i-am-doing-so-please-ignore-missing-odata-id* flag would then allow us to opt-in into failure ignore mode.